doc/specs/vulkan/chapters/memory.txt

```diff
@@ -552,58 +552,70 @@ There must: be at least one memory type with the
 ename:VK_MEMORY_PROPERTY_DEVICE_LOCAL_BIT bit set in its
 pname:propertyFlags.
 
-The memory types are sorted according to a preorder which serves to aid in
-easily selecting an appropriate memory type.
-Given two memory types X and Y, the preorder defines [eq]#X {leq} Y# if:
-
-  * the memory property bits set for X are a strict subset of the memory
-    property bits set for Y.
-    Or,
-  * the memory property bits set for X are the same as the memory property
-    bits set for Y, and X uses a memory heap with greater or equal
-    performance (as determined in an implementation-specific manner).
-
-Memory types are ordered in the list such that X is assigned a lesser
-pname:memoryTypeIndex than Y if [eq]#(X {leq} Y) {land} {lnot} (Y {leq} X)#
-according to the preorder.
-Note that the list of all allowed memory property flag combinations above
-satisfies this preorder, but other orders would as well.
-The goal of this ordering is to enable applications to use a simple search
-loop in selecting the proper memory type, along the lines of:
+For each pair of elements *X* and *Y* returned in pname:memoryTypes, *X*
+must: be placed at a lower index position than *Y* if:
+
+  * either the set of bit flags returned in the pname:propertyFlags member
+    of *X* is a strict subset of the set of bit flags returned in the
+    pname:propertyFlags member of *Y*.
+  * or the pname:propertyFlags members of *X* and *Y* are equal, and *X*
+    belongs to a memory heap with greater performance (as determined in an
+    implementation-specific manner).
+
+[NOTE]
+.Note
+====
+The ordering rules above could allow more than one possible way to order the
+same set of memory types.
```

Comment:
Hmm, I kind of wonder if we should make this more referred to the statement above. Perhaps replacing the first sentence here with something like:

> There is no ordering requirement between memory types where the neither set of bit flags is a subset of the other, allowing more than one possible way to order the same set of memory types in this case.

?

Reply:
Alright, I tried something along these lines.